feature-swap-impl/src/main/java/io/novafoundation/nova/feature_swap_impl/domain/interactor/SwapInteractor.kt

```diff
@@ -29,4 +59,70 @@ class SwapInteractor(
     suspend fun slippageConfig(chainId: ChainId): SlippageConfig? {
         return swapService.slippageConfig(chainId)
     }
+
+    suspend fun validationSystem(chainId: ChainId): SwapValidationSystem? {
+        val assetExchange = assetExchangeFactory.create(chainId) ?: return null
```

Comment:
As we discussed in thre previous PRs - we shouldnt use `assetExchangeFactory` outside of `SwapService`. You can perform a quote using `SwapService` instead